barcode/ean.py

```diff
@@ -90,9 +90,9 @@ def calculate_checksum(self):
         def sum_(x, y):
             return int(x) + int(y)
 
-        evensum = reduce(sum_, self.ean[-2::-2])
-        oddsum = reduce(sum_, self.ean[-1::-2])
-        return (10 - ((evensum + oddsum * 3) % 10)) % 10
+        evensum = reduce(sum_, self.ean[1:12:2])
+        oddsum = reduce(sum_, self.ean[0:11:2])
```

Comment:
This will break some subclasses like `EuropeanArticleNumber8`. You should use `self.digits` instead of `12` here.